README.md
# rhto-GEM: Genome-scale metabolic model of _Rhodotorula toruloides_

[![DOI](https://zenodo.org/badge/133515828.svg)](https://zenodo.org/badge/latestdoi/133515828) [![GitHub version](https://badge.fury.io/gh/sysbiochalmers%2Frhto-gem.svg)](https://badge.fury.io/gh/sysbiochalmers%2Frhto-gem) 

- Brief model description

This repository contains the **updated** genome-scale metabolic model of _Rhodotorula toruloides_, named **rhto-GEM**. For the latest updated release see [here](https://github.com/SysBioChalmers/rhto-GEM/releases).

- Abstract

_Rhodotorula toruloides_ (syn. _Rhodosporidium toruloides_) is a basidiomycetous yeast belonging to the subphylum Pucciniomycotina and occurs naturally in a wide range of habitats including surfaces of leaves, soil and sea water. The broad substrate range of _R. toruloides_ and its ability to accumulate lipids exceeding half of its cell dry weight has made this yeast a popular system for production of biological oils from inedible substrates (e.g. pentose sugars, crude glycerol). The _R. toruloides_ lipid fraction contains ω-3 linolenic acid and heptadecenoic acid, which makes this yeast a promising organism for production of pharma- and nutraceuticals. _R. toruloides_ also produces a number of carotenoid pigments including torularhodin, torulene, γ-carotene and β-carotene. Other applications of _R. toruloides_ include the production of the enzymes L-phenylalanine ammonia-lyase and D-amino acid oxidase.

- Model keywords

**GEM category:** Species; **Utilisation:** experimental data reconstruction; **Field:** metabolic-network reconstruction; **Type of model:** reconstruction, curated; **Model source:** [yeast-GEM](https://github.com/SysBioChalmers/yeast-GEM) & [iYali](https://github.com/SysBioChalmers/Yarrowia_lipolytica_W29-GEM); **Omic source:** genomics; **Taxonomy:** _Rhodotorula toruloides_; **Metabolic system:** general metabolism; **Bioreactor**; **Strain:** NP11; **Condition:** minimal medium;

- Reference:  
> Tiukova IA, Prigent S, Nielsen J, Sandgren M & Kerkhoven EJ (2019) "Genome-scale model of _Rhodotorula toruloides_ metabolism" _Biotechnol Bioeng_. doi:[10.1002/bit.27162](https://onlinelibrary.wiley.com/doi/full/10.1002/bit.27162)


> Lopes HJS _et al_. "C/N ratio and carbon source-dependent lipid production profiling in _Rhodotorula toruloides_" _Appl Microbiol Biotechnol_. doi:[10.1007/s00253-020-10386-5](https://doi.org/10.1007/s00253-020-10386-5)

- Last update: 2020-04-19

- Main model descriptors:

| Taxonomy | Template Model | Reactions | Metabolites | Genes |
| ------------- |:-------------:|:-------------:|:-------------:|:-----:|
| _Rhodotorula toruloides_|	[yeast-GEM](https://github.com/SysBioChalmers/yeast-GEM) & [iYali](https://github.com/SysBioChalmers/Yarrowia_lipolytica_W29-GEM) | 2723 | 2277 | 832 |

A [Memote](https://memote.readthedocs.io/en/latest/) snapshot report of the most recent release is available [here](https://SysBioChalmers.github.io/rhto-GEM).

This repository is administered by [@edkerk](https://github.com/edkerk/), Division of Systems and Synthetic Biology, Department of Biology and Biological Engineering, Chalmers University of Technology.

## Citation

* If you use rhto-GEM in your research, whether for simulations, data analysis, model reconstruction of other purposes, we ask you to cite the rhto-GEM paper on [bioRxiv](https://doi.org/10.1101/528489).
* In addition, it is good practice to cite the specific version of rhto-GEM that you used, to improve reproducibility. All rhto-GEM releases are archived in [Zenodo](https://zenodo.org/badge/latestdoi/133515828). Find the corresponding DOI for each release [here](https://zenodo.org/search?page=1&size=20&q=conceptrecid:2547988&sort=-publication_date&all_versions=True).

## Installation

### Required software

  * This model is recommended to be used with the [**RAVEN toolbox for MATLAB**](https://github.com/SysBioChalmers/RAVEN) (version 2.0).
  * Alternatively, the model can also be directly used with the [COBRA toolbox for MATLAB](https://github.com/opencobra/cobratoolbox), [cobrapy](https://github.com/opencobra/cobrapy), or other software or toolboxes that support genome-scale models in SBML L3V1 FBCv2 format.

### Dependencies
* Please see the [RAVEN toolbox](https://github.com/SysBioChalmers/RAVEN) repository for dependencies regarding RAVEN.
* For contribution to development: a [git wrapper](https://github.com/manur/MATLAB-git) added to the search path.

### Installation instructions
* Just want to use the model? Clone it from [`master`](https://github.com/SysBioChalmers/rhto-GEM) in the Github repo, or just download [the latest release](https://github.com/SysBioChalmers/rhto-GEM/releases).
* Wish to also contribute? Fork it to your Github account, and create a new branch from [`devel`](https://github.com/SysBioChalmers/rhto-GEM/tree/devel).

## Model files

The model is available in `.xml`, `.txt`, `.yml`, `.mat` and `.xlsx` (the last 2 extensions only in `master`). Additionally, versions of toolboxes & SBML used for saving the model are tracked in `dependencies.txt`.

### Complementary scripts

* `newCommit.m`: prepares files from a modified model for a new GitHub commit in a development branch.
* `newCommit.m`: prepares files from a modified model for a new GitHub release in the master branch.
* `Lopes2019`: folder with scripts related to Lopes _et al_. (2019)
* `analysis`: folder with scripts performing analyses on rhto-GEM.
* `curation`: folder with additional curation scripts, after the initial reconstruction
* `experimental`: folder with scripts that modify _rhto-GEM_ to incorporate experimental data.
* `reconstruction`: folder with scripts that detail the model reconstruction and curation process.
* `validation`: folder with scripts validating performance of rhto-GEM.

### Complementary data

* `Lopes2019`: folder with data related to Lopes _et al_. (2019)
* `data`: experimentally measured data.
* `genome`: protein fasta files of _R. toruloides_, _S. cerevisiae_ and _Y. lipolytica_, as used for identifying orthologs.
* `meneco`: input and output files for meneco, as run for `r4_gapfilling`.
* `reconstruction`: miscellaneous files used during model reconstruction.
* `validation`: experimental data used to validate _rhto-GEM_.

## Contributors
* [Ievgeniia Tiukova](https://www.chalmers.se/en/staff/Pages/tiukova.aspx) ([@itaova](https://github.com/itaova)), Chalmers University of Technology, Gothenburg Sweden
* [Eduard J. Kerkhoven](https://www.chalmers.se/en/staff/Pages/Eduard-Kerkhoven.aspx) ([@edkerk](https://github.com/edkerk)), Chalmers University of Technology, Gothenburg Sweden


history.md
# History

### rhto v1.3.0:
* Fix
  * Remove incorrect isoenzymes by manual curation of grRules (PR #44)
  * Remove duplicate transport reactions (PR #45)
  * Correct EC codes of mitochondrial 3-ketoacyl-CoA thiolases (PR #45)
* Documentation
  * Update citation and model stats in README.md

### rhto v1.2.2:
* Features
  * Data and analysis complementary to [Lopes _et al._, 2020, Appl Microbiol Biotechnol](https://doi.org/10.1007/s00253-020-10386-5).  (PR #38).
* Documentation
  * Update citation of _rhto-GEM_ paper, now available from Biotechnology & Bioengineering (PR #39).

### rhto v1.2.1:
* Features
  * Curate grRules and existence of several reactions based on [doi:10.1186/s12934-015-0217-5](https://doi.org/10.1186/s12934-015-0217-5) (PR #34).
  * Curate grRules of xylose and glucose transporters based on differential expression in proteomics study [doi:10.1186/s13068-019-1478-8](https://doi.org/10.1186/s13068-019-1478-8) (PR #34).
* Fix
  * Bug in `sumBioMass` resulted in incorrect fitted carbohydrate fraction of biomass equation, also resulting in incorrect NGAM (PR #33).
  * Remove four duplicate reactions (`r_4235`, `y200008`, `r_1000`, `r_4262`) (PR #34).
* Refactor
  * Standardize two complex gene associations (`r_0831`, `r_0832`) (PR #34).

### rhto v1.2.0:
* Features
  * Set alternative carbon source exchange reactions as reversible (glycerol, xylose) (PR #28).
  * `init_rhto-GEM.m` script that prepares MATLAB environment with path to repo's folders (PR #28).
  * `makeLipidRxns` now takes list of acyl-chain configurations, instead of defining these by itself from a list of allowed acyl-chains.  (PR #28).
  * Correct gene associations of mitochondrial beta-oxidation reaction (PR #29).
  * Curate subsystems and EC numbers (PR #29).
  * Update DNA, RNA and amino acid pseudoreactions, based on frequency of nucleotides and amino acids in genome, transcript and protein FASTA files (PR #30).
  * Scale carbohydrate fraction to ensure that biomass pseudoreaction generates 1 gDCW (PR #30).
  * Refit energetics (PR #30).
* Fix
  * Curate acyl-chains: simplify number of possible chain combinations, based on measured chain distribution (e.g. 16:1 level is around 1% and therefore not included in the biomass), and distribution of acyl-chains on TAGs ([10.1007/s00253-017-8126-7](http://doi.org/10.1007/s00253-017-8126-7)):  sn-1 is 16:0 or 18:0, sn-2 is 18:1, 18:2 or 18:3; sn-3 is 16:0, 18:0 or 18:1. (PR #28).
  * Use template SLIMER reactions (PR #28).
  * Various curations of lipid metabolism (PR #28, PR #30).

### rhto v1.1.1:
* Features
  * Make Memote snapshot report via Travis with every push to `master`, report available at https://sysbiochalmers.github.io/rhto-GEM/ (PR #23).
  * Sort reactions and metabolites when saving model, for easier `diff` (PR #23).
  * Include phosphoketolase and phosphate transacetalyase reactions (PR #23).
  * Increase annotation coverage for metabolites and reactions by mapping to MetaNetX (PR #23).
  * Script for gene essentiality validation (PR #23).
* Fix
  * Remove duplicated reactions, introduced at various steps of reconstruction (PR #23).
  * Set fatty-CoA-ligases as irreversible, as they could otherwise form ATP regenerating loops that are unlikely to appear _in vivo_ (PR #23).
  * Refit energetics with new `NGAM` and `GAM` values (PR #23).
* Refactor
  * Change style of graph with growth rate validation (PR #23).
  * Refactor various simulation scripts (PR #23).
  * All new (non `yeast-GEM`) reactions are assigned identifiers with prefix `t_`  (PR #23).

### rhto v1.1.0:
* Features
  * Include model of _Yarrowia lipolytica_ [iYali4.1.1](https://github.com/SysBioChalmers/Yarrowia_lipolytica_W29-GEM/releases/tag/4.1.1) as template during reconstruction (PR #18).
  * Standardize identifiers: all new metabolites have ids `m_xxxx`, while new reactions are `t_xxxx` or `tr_xxxx` (for transport reactions) (PR #18).
  * Reduced the amount of hard-coded ids in scripts (PR #18).
* Fix
  * Include missing reactions such as complex I (PR #18).
  * Re-fit energetics (NGAM and GAM) (PR #18).
* Reorder
  * Reorganize reconstruction process: use TSV files for lists of curations (PR #18).
  * Include legend in validation graph (PR #18).

### rhto v1.0.0:
* Documentation
  * Update readme.md to include correct links to Zenodo and bioRxiv (PR #15).

### rhto v0.0.2:
* Features
  * Include script `ComplementaryScripts\validation\growthPrediction.m` to valide model by comparing growth predictions (PR #13).
  * Include script `ComplementaryScripts\analysis\fseof_target_prediction.m` for FSEOF target prediction (PR #13).
* Fixes
  * Energetics script `r7_energetics.m` correctly reads file with flux data (PR #13).
  * Correct unlogical and duplicate grRules (PR #13).
  * Correct subSystems naming, removing orphan indices from template model (PR #13).
* Documentation
  * `Readme.md` has detailed text on how to cite the model (PR #13).
  * `Readme.md` contains version badge and Zenodo doi badge (PR #13).

### rhto v0.0.1:
Initial release of rhto-GEM.

ComplementaryScripts/curation/curation_c1.m
clear;clc;if ~exist('scripts') | ~endsWith(scripts,'ComplementaryScripts'); run('../../init_rhtoGEM.m'); end
% This script takes rhto-GEM v1.2.0 and performs curation based on e.g.
% literature data
model = importModel([root '/ModelFiles/xml/rhto.xml']);

% Set default UB and LB
model.annotation.defaultUB = 1000;
model.annotation.defaultLB = -1000;
model.lb(isinf(model.lb)) = -1000;
model.ub(isinf(model.ub)) = 1000;

%% Curate glucose and xylose transporters
% From proteomics measurements of cultivations on glucose and xylose, infer
% which genes code for transporters (https://doi.org/10.1186/s13068-019-1478-8).
model = changeGrRules(model,'r_1166','RHTO_06080 or RHTO_01923 or RHTO_04266 or RHTO_00984 or RHTO_06016 or RHTO_03685 or RHTO_05157',true);
model = changeGrRules(model,'r_1719','RHTO_01630 or RHTO_03448 or RHTO_07444 or RHTO_06801 or RHTO_06080 or RHTO_01923',true);
rxnIDs = getIndexes(model,{'r_1166','r_1719'},'rxns');
model.rxnNotes(rxnIDs) = {'Differential proteomics expression'};
model.rxnReferences(rxnIDs) = {'doi:10.1186/s13068-019-1478-8'};

%% Curate gene association based on previous reconstruction
model = changeGrRules(model, 'r_0300', 'RHTO_07345', true);
model = changeGrRules(model, 'r_0301', 'RHTO_06406', true);
model = changeGrRules(model, 'r_0545', 'RHTO_06717', true);
model = changeGrRules(model, 'r_0658', 'RHTO_01289 or RHTO_01290', true);
model = changeGrRules(model, 'r_1094', 'RHTO_04556', true);
model = changeGrRules(model, 'r_0205', 'RHTO_00641 or RHTO_07387 ', true);
model = changeGrRules(model, 'r_0454', 'RHTO_05714', true);
model = changeGrRules(model, 'r_0455', 'RHTO_01560', true);
rxnIDs = getIndexes(model,{'r_0300','r_0301','r_0545','r_0658','r_1094','r_0205','r_0454','r_0455'},'rxns');
model.rxnNotes(rxnIDs) = {'Manual curation'};
model.rxnReferences(rxnIDs) = {'doi:10.1186/s12934-015-0217-5'};

% Two additional reactions were identified
metsToAdd.metNames      = {'L-xylulose'};
metsToAdd.compartments  = {'c'};
metsToAdd.metFormulas   = {'C5H10O5'};
metsToAdd.mets          = generateNewIds(model,'mets','m_',length(metsToAdd.metNames));
model                   = addMets(model,metsToAdd); clear metsToAdd;

rxnsToAdd.equations     = {'L-arabinitol[c] + NAD[c] <=> L-xylulose[c] + NADH[c] + H+[c]', 'L-xylulose[c] + NADPH[c] + H+[c] <=> xylitol[c] + NADP(+)[c]'};
rxnsToAdd.rxnNames      = {'arabinitol 4-dehydrogenase', 'L-xylulose reductase'};
rxnsToAdd.grRules       = {'RHTO_01629','RHTO_00373'};
rxnsToAdd.subSystems    = {{'Pentose and glucuronate interconversions'},{'Pentose and glucuronate interconversions'}};
rxnsToAdd.eccodes       = {'1.1.1.12','1.1.1.10'};
rxnsToAdd.rxns          = generateNewIds(model,'rxns','t_',length(rxnsToAdd.rxnNames));
rxnsToAdd.rxnNotes      = {'Manual curation', 'Manual curation'};
rxnsToAdd.rxnReferences = {'doi:10.1186/s12934-015-0217-5', 'doi:10.1186/s12934-015-0217-5'};
model                   = addRxns(model,rxnsToAdd,3,'',false,true); clear rxnsToAdd

%% Various corrections
% Duplicate reactions
model = removeReactions(model,{'r_4235', 'y200008', 'r_1000', 'r_4262'}, true, true);

% Rephrased grRules, removing unclear gene associations
model = changeGrRules(model, 'r_0831', '(RHTO_02312 and RHTO_07893 and RHTO_04225) or (RHTO_02312 and RHTO_07893 and RHTO_07860)', true);
model = changeGrRules(model, 'r_0832', '(RHTO_02312 and RHTO_07893 and RHTO_04225) or (RHTO_02312 and RHTO_07893 and RHTO_07860)', true);

%%
save([root '/scrap/model_c1.mat'])
cd ..; newCommit(model); cd curation
